code/modules/recycling/package_wrapper.dm

```diff
@@ -0,0 +1,174 @@
+///Returns the amount of sheets of wrapping paper that the given object would need to be wrapped.
+/proc/wrapping_paper_needed_for_obj(var/obj/O)
+	if(istype(O))
+		return min(O.w_class, 8) //At most, since 20 sheets is a bit much for structures
+
+	else if(ismob(O))
+		//Mob size is scaled weirdly, so we have to slap it through a switch
+		var/mob/M = O
+		var/nb_sheets = 1
+		switch(M.mob_size)
+			if(MOB_SIZE_LARGE)
+				nb_sheets = 15
+			if(MOB_SIZE_MEDIUM)
+				nb_sheets = 10
+			if(MOB_SIZE_SMALL)
+				nb_sheets = 5
+			if(MOB_SIZE_TINY)
+				nb_sheets = 2
+		return nb_sheets
+	
+	CRASH("Tried to get the amount of paper sheets needed to wrap invalid object type '[O? O.type : "null"]'.")
+
+///////////////////////////////////////////////////////////////////////////////////////
+// Parcel Wrapper
+///////////////////////////////////////////////////////////////////////////////////////
+/obj/item/stack/package_wrap
+	name             = "package wrapper roll"
+	desc             = "Heavy duty brown paper used to wrap packages to protect them during shipping."
+	icon             = 'icons/obj/items/gift_wrapper.dmi'
+	icon_state       = "deliveryPaper"
+	singular_name    = "sheet"
+	w_class          = ITEM_SIZE_NORMAL
+	max_amount       = 50
+	material         = /decl/material/solid/paper
+	matter           = list(/decl/material/solid/cardboard = MATTER_AMOUNT_PRIMARY * HOLLOW_OBJECT_MATTER_MULTIPLIER) //For the tube
+	matter_per_piece = list(/decl/material/solid/paper = MATTER_AMOUNT_PRIMARY)
+	force            = 1
+	throwforce       = 1
+	throw_range      = 5
+	throw_speed      = 3
+	///Types that the wrapper cannot wrap, ever
+	var/static/list/wrap_blacklist = list(
```

Comment:
I would really suggest making this a proc so that it can be modified by subtypes. Static vars defined on types can't be overridden on subtypes, while if you did something like
```
/obj/item/stack/package_wrap/proc/get_wrap_blacklist()
	var/static/list/wrap_blacklist = list(
		/obj/item/parcel,
		/obj/item/parcel/gift,
		/obj/item/evidencebag,
		/obj/item/stack/package_wrap,
	)
	return wrap_blacklist
```
you could then do
```
/obj/item/stack/package_wrap/recursive/get_wrap_blacklist()
	var/static/list/wrap_blacklist = (..() - list(/obj/item/parcel, /obj/item/parcel/gift, /obj/item/stack/package_wrap))
	return wrap_blacklist
```
and modify it on a subtype. This also late-initialises the list, and ensures it's only modified/created once. (Make sure nothing mutates it, though!!)

Reply:
Yeah, that's a good idea. I'll try that.